proxy/src/fully_qualified_authority.rs

```diff
@@ -5,9 +5,26 @@ use std::str::FromStr;
 
 use http::uri::Authority;
 
+/// A fully qualified authority according to Kubernetes service naming
+/// conventions.
```

Comment:
Actually, this isn't restricted to things that are named according to Kuberentes conventions. I suggest something like "A normalized `Authority`."